src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -49,6 +58,30 @@ public PersonCard(ReadOnlyPerson person, int displayedIndex) {
     }
 
     /**
+     * Gets a random unused color for the new tagName, or returns the corresponding color of the old tagName
+     * @param tagName
+     * @return the color of the tag
+     */
+    private String getTagColor(String tagName) {
+        if (!tagColors.containsKey(tagName)) {
+            tagColors.put(tagName, COLORS[colorIndex]);
+            updateColorIndex();
+        }
+        return tagColors.get(tagName);
+    }
```

Comment:
This method abstraction is problematic. Its name sounds like it's simply doing a retrieval but it is actually changing the state of the data.